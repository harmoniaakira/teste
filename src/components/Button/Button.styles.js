import styled from 'styled-components'

export const ButtonComponent = styled.button`
  width: 280px;
  min-height: 48px;
  font-size: 12px;
  line-height: 24px;
  font-weight: 700;
  border: none;
  border-radius: 100px;
  box-shadow: 0px 6px 10px rgba(0, 0, 0, 0.11);
  transition: background-color 0.5s ease;
  text-transform: uppercase;
  cursor: pointer;

  ${props => props.fullScreen 
    ? "margin: 0 auto;"
    : ""
  }

  ${props => props.disabled 
    ? "background-color: #E0E0E0;" +
      "color: #8E8E93;" +
      "cursor: default;"
    : "background-color: #054375;" +
      "color: white;"
  }

  &:hover {
    ${props => props.disabled 
      ? "background-color: #E0E0E0;"
      : "background: #043C64;"
    }
  }

  @media (min-width: 1440px) {
    width: 158px;
    float: right;
  }
`
export const ButtonPopup = styled.button`
  width: 280px;
  height: 48px;
  font-size: 14px;
  line-height: 16px;
  font-weight: 600;
  letter-spacing: 0.75px;
  border: none;
  border-radius: 100px;
  text-transform: uppercase;
  transition: background-color 0.5s ease;
  cursor: pointer;
  background-color: transparent;

  ${props => props.disabled 
    ? "color: #8E8E93;" +
      "cursor: default;"
    : "color: #054375;"
  }

  &:hover {
    ${props => props.disabled 
      ? "background-color: transparent;"
      : "background: rgba(5, 67, 117, .1);"
    }
  }

  @media (min-width: 1440px) {
    width: 158px;
    float: right;
  }
`