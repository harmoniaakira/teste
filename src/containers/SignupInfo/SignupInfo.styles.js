import styled from 'styled-components'
import IconButtonComponent from '@material-ui/core/IconButton';

export const Container = styled.div`
  text-align: left;
  margin: 0 auto 60px auto;
  width: 100%;
  max-width: 768px;
  padding: 0 24px;

  @media (min-width: 1440px) {
    max-width: 1440px;
    margin: 0;
    text-align: unset;
    padding: 0;
  }
`

export const Content = styled.div`
  width: 518px;
  min-height: 170px;
  max-height: 928px;
  margin: 0;
  position: absolute;
  top: 24%;
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
`

export const Avatar = styled.img`

`

export const IconButton = styled(IconButtonComponent)`
  color: #8E8E93;
`

export const Breadcrumb = styled.div `
  display: flex;
`

export const BreadcrumbTitle = styled.div `
  color: #636366;
  font-style: normal;
  font-weight: bold;
  font-size: 20px;
  line-height: 150%;
  margin: auto 0;
`