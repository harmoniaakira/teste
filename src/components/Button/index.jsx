import React from 'react'
import PropTypes from 'prop-types';
import { ButtonComponent, ButtonPopup } from './Button.styles'

const Button = ( { handleClick, disabled, children, popup } ) => {
    
	return ( popup 
		? 	<ButtonPopup 
				onClick={handleClick}
				disabled={disabled}
			>
				{children}
			</ButtonPopup>
		:	<ButtonComponent 
				onClick={handleClick}
				disabled={disabled}
			>
				{children}
			</ButtonComponent>
	)
}

Button.propTypes = {
	handleClick: PropTypes.func.isRequired,
	children: PropTypes.oneOfType([
        PropTypes.arrayOf(PropTypes.node),
        PropTypes.node
    ]).isRequired,
	disabled: PropTypes.bool,
	popup: PropTypes.bool
	
}

Button.defaultProps = {
	disabled: false,
	popup: false
}

export default Button
