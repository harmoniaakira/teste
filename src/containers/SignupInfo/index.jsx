import ArrowBack from '@material-ui/icons/ArrowBackIos';
import Button from 'components/Button';
import { MobileView } from 'components/Layout/Layout.styles';
import Web from 'components/Web';
import React from 'react';
import { useHistory } from 'react-router-dom';
import {
  Breadcrumb,
  BreadcrumbTitle, Container,
  Content,
  IconButton
} from './SignupInfo.styles';

const SignupInfo = () => {
  const history = useHistory();

  const handleClick = () => {
    history.goBack();
  }

  return (
    <Container>
      <MobileView>
        <Breadcrumb>
          <IconButton
            edge="start"
            color="inherit"
            onClick={handleClick}
            aria-label="close"
          >
            <ArrowBack />
          </IconButton>
          <BreadcrumbTitle>
            Preenchimento de dados
          </BreadcrumbTitle>
        </Breadcrumb>
        <br /><br />
        <strong>CPF</strong>
        <br />
        Insira somente os 11 dígitos do CPF do cliente. Ex.: 000.000.000-00
        <br /><br />
        <strong>Data de Nascimento</strong>
        <br />
        <span>Preencha a Data de Nascimento do cliente, igual ao do documento de identificação. Ex.: 00/00/0000</span>
        <br /><br />
        <strong>Número de Celular válido</strong>
        <br />
        <span>É necessário informar um número de celular válido. Ex.: (00) 00000-0000</span>
        <br /><br />
        <strong>E-mail principal válido</strong>
        <span>Importante informar o e-mail principal que é utilizado e também que seja válido. Ex.: contatodamaria@gmail.com</span>
        <br /><br />
      </MobileView>

      <Web>
        <Content>
          <Breadcrumb>
            <IconButton
              edge="start"
              color="inherit"
              onClick={handleClick}
              aria-label="close"
            >
              <ArrowBack />
            </IconButton>
            <BreadcrumbTitle>
              Preenchimento de dados
            </BreadcrumbTitle>
          </Breadcrumb>
          <br /><br />
          <strong>CPF</strong><br />
          Insira somente os 11 dígitos do CPF do cliente. Ex.: 000.000.000-00
          <br /><br />
          <strong>Data de Nascimento</strong><br />
          <span>Preencha a Data de Nascimento do cliente, igual ao do documento de identificação. Ex.: 00/00/0000</span>
          <br /><br />
          <strong>Número de Celular válido</strong><br />
          <span>É necessário informar um número de celular válido. Ex.: (00) 00000-0000</span>
          <br /><br />
          <strong>E-mail principal válido</strong><br />
          <span>Importante informar o e-mail principal que é utilizado e também que seja válido. Ex.: contatodamaria@gmail.com</span>
          <br /><br />
          <Button handleClick={handleClick}>Voltar</Button>
        </Content>
      </Web>
    </Container>
  )
}

export default SignupInfo;
