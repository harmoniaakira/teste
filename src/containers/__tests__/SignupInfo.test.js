import React from 'react'
import { shallow, render } from 'enzyme'
import SignupInfo from 'containers/SignupInfo'
import Button from 'components/Button';

const mockHistoryBack = jest.fn();
jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useHistory: () => ({
        goBack: mockHistoryBack,
    }),
}));

describe("SignupInfo", () => {
    const signupInfo = shallow(<SignupInfo />);

    it('should call history goback', () => {
        signupInfo.find(Button).props().handleClick();
        expect(mockHistoryBack.mock.calls.length).toEqual(1);
    })

    it('should match snapshot', () => {
        expect(signupInfo).toMatchSnapshot()
    })
})