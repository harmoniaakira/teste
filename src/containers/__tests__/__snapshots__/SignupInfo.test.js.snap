// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignupInfo should match snapshot 1`] = `
<styled.div>
  <styled.div>
    <styled.div>
      <Styled(WithStyles(ForwardRef(IconButton)))
        aria-label="close"
        color="inherit"
        edge="start"
        onClick={[Function]}
      >
        <Memo(ForwardRef(ArrowBackIosIcon)) />
      </Styled(WithStyles(ForwardRef(IconButton)))>
      <styled.div>
        Preenchimento de dados
      </styled.div>
    </styled.div>
    <br />
    <br />
    <strong>
      CPF
    </strong>
    <br />
    Insira somente os 11 dígitos do CPF do cliente. Ex.: 000.000.000-00
    <br />
    <br />
    <strong>
      Data de Nascimento
    </strong>
    <br />
    <span>
      Preencha a Data de Nascimento do cliente, igual ao do documento de identificação. Ex.: 00/00/0000
    </span>
    <br />
    <br />
    <strong>
      Número de Celular válido
    </strong>
    <br />
    <span>
      É necessário informar um número de celular válido. Ex.: (00) 00000-0000
    </span>
    <br />
    <br />
    <strong>
      E-mail principal válido
    </strong>
    <span>
      Importante informar o e-mail principal que é utilizado e também que seja válido. Ex.: contatodamaria@gmail.com
    </span>
    <br />
    <br />
  </styled.div>
  <Web>
    <styled.div>
      <styled.div>
        <Styled(WithStyles(ForwardRef(IconButton)))
          aria-label="close"
          color="inherit"
          edge="start"
          onClick={[Function]}
        >
          <Memo(ForwardRef(ArrowBackIosIcon)) />
        </Styled(WithStyles(ForwardRef(IconButton)))>
        <styled.div>
          Preenchimento de dados
        </styled.div>
      </styled.div>
      <br />
      <br />
      <strong>
        CPF
      </strong>
      <br />
      Insira somente os 11 dígitos do CPF do cliente. Ex.: 000.000.000-00
      <br />
      <br />
      <strong>
        Data de Nascimento
      </strong>
      <br />
      <span>
        Preencha a Data de Nascimento do cliente, igual ao do documento de identificação. Ex.: 00/00/0000
      </span>
      <br />
      <br />
      <strong>
        Número de Celular válido
      </strong>
      <br />
      <span>
        É necessário informar um número de celular válido. Ex.: (00) 00000-0000
      </span>
      <br />
      <br />
      <strong>
        E-mail principal válido
      </strong>
      <br />
      <span>
        Importante informar o e-mail principal que é utilizado e também que seja válido. Ex.: contatodamaria@gmail.com
      </span>
      <br />
      <br />
      <Button
        disabled={false}
        handleClick={[Function]}
        popup={false}
      >
        Voltar
      </Button>
    </styled.div>
  </Web>
</styled.div>
`;
